### PR TITLE
Fix remapped values in public dashboards

### LINKED ISSUE #44231: Field values remapping with mixed PK and a FK targeting the same PK is broken in public dashboards
### Describe the bug

https://metaboat.slack.com/archives/C01LQQ2UW03/p1718352501337109

Remapped field values are not shown in public dashboards when a parameter is mapped to a FK field.

#### Bug 1

When:
- A parameter is mapped to a PK and to a FK targeting this PK (different questions)
- In the PK table there is a field with the semantic type set to `type/Name`
- For the PK column `has_field_values` is set to `list`

Then:
- Parameter values are not shown in public dashboards

#### Bug 2

When:
- Same as 1, but set `has_field_values` for the FK field to `list` as well

Then:
- Dropdown values would not appear even in regular dashboards

### Expected behavior

- The parameter values endpoint should not throw an exception because it cannot combine remapped and non-remapped values. Values from multiple fields should be merged in one list.
- It should be possible to use remapped values in public dashboards as in the app.

### Logs

- 

### Information about your Metabase installation

```JSON
v50
```

### Severity

P1/P2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/44231

2 issues:
- mixed remapped & non-remapped values were not handled in the parameter values call (exception - malli schema error)
- foreign key `target` field was not hydrated in public dashboards; this case worked in the app only because `target` was hydrated in `query_metadata`

BE changes implemented by @qnkhuat

### CODE REVIEW

File: src/metabase/models/params.clj [hunk 1]
```diff
@@ -110,7 +110,7 @@
   just the column identifiers, perhaps for use with something like `select-keys`. Clutch!
 
     (t2/select Field:params-columns-only)"
-  ['Field :id :table_id :display_name :base_type :semantic_type :has_field_values])
+  ['Field :id :table_id :display_name :base_type :semantic_type :has_field_values :fk_target_field_id])
```
Comment (qnkhuat) [inline src/metabase/models/params.clj:113]: @ranquild, why do you need this for the public dashboard to work?

File: src/metabase/api/dashboard.clj [hunk 2]
```diff
@@ -997,6 +997,20 @@
                                               1 first)))
          :has_more_values has_more_values}))))
 
+(defn- combine-chained-fitler-results
```
Comment (noahmoss) [inline src/metabase/api/dashboard.clj:1000]:
```suggestion
(defn- combine-chained-filter-results
```

File: src/metabase/api/dashboard.clj [hunk 3]
```diff
@@ -997,6 +997,20 @@
...
+  [results]
+  (let [;; merge values with remapped values taking priority
+        values (->> (mapcat :values results)
+                    (sort-by count)
+                    (m/index-by first)
+                    vals)]
+    (cond->> values
+      (seq values)
+      ;; sort by remapped values only if all values are remapped
```
Comment (noahmoss) [inline src/metabase/api/dashboard.clj:1009]: is this accurate if it's only looking at the first value? the comment sounds like it looks at every value.

Comment (noahmoss) [reply src/metabase/api/dashboard.clj:1009]: Something like `(sort-by #(or (second %) (first %))` might be closer to the behavior described. i.e. `Sort by remapped value if it exists`

File: test/metabase/api/dashboard_test.clj [hunk 4]
```diff
@@ -402,6 +402,25 @@
                             :dashboard_id       dashboard-id}]
           (is (#'api.dashboard/get-dashboard dashboard-id)))))))
 
+(deftest get-dashboard-param-fields-has-target-test
+  (testing "param-fields for fk has target (#44231)"
+    (mt/with-temp
+      [:model/Card          {orders-card-id :id}   {:database_id   (mt/id)
```
Comment (noahmoss) [inline test/metabase/api/dashboard_test.clj:408]: whitespace nit (or align them)
```suggestion
      [:model/Card          {orders-card-id :id} {:database_id   (mt/id)
```

File: test/metabase/api/dashboard_test.clj [hunk 5]
```diff
@@ -2960,6 +2979,48 @@
...
+      :model/DashboardCard _             {:card_id            products-card-id
+                                          :dashboard_id       dash-id
+                                          :parameter_mappings [{:parameter_id "__ID__"
+                                                                :card_id      products-card-id
+                                                                :target       [:dimension (mt/$ids products $id)]}]}]
+     (mt/with-column-remappings [products.id products.title]
+       (is (=? {:values         [[(mt/malli=? pos-int?) "Aerodynamic Bronze Hat"]
+                                 [(mt/malli=? pos-int?) "Aerodynamic Concrete Bench"]
+                                 [(mt/malli=? pos-int?) "Aerodynamic Concrete Lamp"]]
+                :has_more_values false}
```
Comment (noahmoss) [inline test/metabase/api/dashboard_test.clj:3010]:
```suggestion
       (is (=? {:values          [[(mt/malli=? pos-int?) "Aerodynamic Bronze Hat"]
                                  [(mt/malli=? pos-int?) "Aerodynamic Concrete Bench"]
                                  [(mt/malli=? pos-int?) "Aerodynamic Concrete Lamp"]]
                :has_more_values false}
```

**Review by noahmoss — APPROVED** [review]

File: src/metabase/api/dashboard.clj (hunk 3 above)
Comment (ranquild) [reply src/metabase/api/dashboard.clj:1009]: that's not the same - as we sort the rows the first value would have 2 items only if all are remapped. Your code would do it differently

**Pull request merged**